### PR TITLE
Update processing documentation with code from redesign

### PR DESCRIPTION
**Pull request by ngkavin**
New code blocks that work with the current redesigned package replaced the old code.

### CODE REVIEW

File: docs/source/process.rst [hunk 1]
```diff
@@ -22,75 +21,40 @@ Functionality
...
-   ed.calibrate(save=True, save_postfix='_Cal')
-
-   # Save output to another directory: ./cal_results/convertedfile_Sv.nc
-   ed.calibrate(save=True, save_path='./cal_results')
-
-   # Save output to another directory with an arbitrary name
-   ed.calibrate(save=True, save_path='./cal_results/somethingnew.nc')
+   import echopype as ep
+   nc_path = './converted_files/file.nc'           # path to a converted nc file
+   echodata = ep.open_converted(nc_path)           # create a processing object
```
Comment (emiliom) [inline docs/source/process.rst:37]: > create a processing object

That's no longer a `processing` object, per se. Change to "create an EchoData object"

File: docs/source/process.rst [hunk 2]
```diff
@@ -22,75 +21,40 @@ Functionality
...
+   MVBS = ep.preprocess.get_MVBS(Sv_clean)         # obtain MVBS from denoised Sv
 
-By default, for noise removal and MVBS calculation, echopype tries to load Sv
-already stored in memory (``ed.Sv``), or tries to calibrate the raw data to
-obtain Sv. If ``ed.Sv`` is empty (i.e., whe calibration operation has not been
-performed on the object), echopype will try to load Sv from ``*_Sv.nc`` from
-the directory containing the converted ``.nc`` file or from the user-specified
-path. For example:
+The functions in the ``calibrate`` subpackage take in an ``EchoData`` object,
+which is essentially a container for multiple xarray ``Dataset`` groups,
```
Comment (emiliom) [inline docs/source/process.rst:43]: Change to "which is essentially a container for multiple xarray ``Datasets``,"

File: docs/source/process.rst [hunk 3]
```diff
@@ -22,75 +21,40 @@ Functionality
...
-   # Save output to another directory: ./cal_results/convertedfile_Sv.nc
-   ed.calibrate(save=True, save_path='./cal_results')
-
-   # Save output to another directory with an arbitrary name
-   ed.calibrate(save=True, save_path='./cal_results/somethingnew.nc')
+   import echopype as ep
+   nc_path = './converted_files/file.nc'           # path to a converted nc file
+   echodata = ep.open_converted(nc_path)           # create a processing object
+   Sv = ep.calibrate.compute_Sv(echodata)          # obtain Sv
+   Sv_clean = ep.preprocess.remove_noise(Sv, ...)  # obtain a denoised Sv Dataset
```
Comment (emiliom) [inline docs/source/process.rst:39]: What are those 3 dots?

File: docs/source/process.rst [hunk 4]
```diff
@@ -115,33 +79,29 @@ By default, echopype uses the following for calibration:
...
-
-This will trigger recalculation of all related parameters,
-including sound speed, seawater absorption, thickness of each sonar
-sample, and range. The updated values can be retrieved with:
+   environment = {
+       'temperature': 8,   # temperature in degree Celsius
+       'salinity': 30,      # salinity in PSU
+       'pressure': 50,     # pressure in dbar
+   }
+   ep.calibrate.compute_Sv(env_params=environment)
```
Comment (emiliom) [inline docs/source/process.rst:99]: This looks incomplete. Compare it to the example above: `Sv = ep.calibrate.compute_Sv(echodata)`

File: docs/source/process.rst [hunk 5]
```diff
@@ -115,33 +79,29 @@ By default, echopype uses the following for calibration:
...
 
-.. code-block:: python
-
-   ed.seawater_absorption  # absorption in [dB/m]
-   ed.sound_speed          # sound speed in [m/s]
-   ed.sample_thickness     # sample spatial resolution in [m]
-   ed.range                # range for each sonar sample in [m]
+These value will be used in calculating sound speed,
+seawater absorption, thickness of each sonar sample, and range.
```
Comment (emiliom) [inline docs/source/process.rst:102]: I think it'd be helpful to show here how those calculated variables can be accessed. Include a brief code block analogous to what was there before.

File: docs/source/process.rst [hunk 6]
```diff
@@ -165,24 +125,31 @@ from the raw data file.
...
+specify ``cal_params`` while calling the calibration functions like so:
 
 .. code-block:: python
 
-   ed.equivalent_beam_angle.loc[dict(frequency=18000)] = -18.02  # set value for 18 kHz only
+   # set all channels at once
+   calibration = {
+       'equivalent_beam_angle': [-17.47, -20.77, -21.13, -20.4 , -30]
+   }
+   ep.calibrate.compute_Sv(cal_params=calibration)
```
Comment (emiliom) [inline docs/source/process.rst:144]: This looks incomplete. Compare it to the example above: `Sv = ep.calibrate.compute_Sv(echodata)`

**Review by emiliom — COMMENTED** [review]
Thanks @ngkavin! I added a bunch of small comments.

File: docs/source/process.rst (hunk 3 above)
Comment (ngkavin) [reply docs/source/process.rst:39]: This function requires additional parameters to be passed in. Should I explicitly write down the required arguments?

File: docs/source/process.rst (hunk 5 above)
Comment (ngkavin) [reply docs/source/process.rst:102]: Since these are just function that output an xarray `Dataset`, accessing these values would require creating objects that are not exposed to the user via the `__init__.py` of `calibrate`. So, I don't think you can access these values with the current API.

File: docs/source/process.rst (hunk 3 above)
Comment (emiliom) [reply docs/source/process.rst:39]: Ah.If they are only a couple or so, my suggestion is to include them with reasonable values. Otherwise, I guess leaving the ellipsis is ok.

File: docs/source/process.rst (hunk 5 above)
Comment (ngkavin) [reply docs/source/process.rst:106]: After further investigation, these are provided in the output dataset. I'll update the docs.

**Comment by emiliom** [comment]
Thanks for the changes. Looks good to me now. @leewujung please merge after reviewing -- or just go ahead and merge as is if you prefer.

File: docs/source/process.rst [hunk 7]
```diff
@@ -22,75 +21,44 @@ Functionality
 
 - EK80 and EA640 broadband echosounders:
 
-  - calibration based on pulse compression output in the
+  - Calibration based on pulse compression output in the
```
Comment (leewujung) [inline docs/source/process.rst:24]: Please add `(beta: we found inconsistency between pulse compression outputs from the current echopype implement, EchoView, and Matlab EchoLab code, see #308 )` at the end. Make sure the issue is linked correctly.

File: docs/source/process.rst [hunk 8]
```diff
@@ -2,18 +2,17 @@ Data processing
...
 Functionality
 -------------
 
 - EK60 and AZFP narrowband echosounders:
 
-  - calibration and echo-integration to obtain
+  - Calibration and echo-integration to obtain
     volume backscattering strength (Sv) from power data.
   - Simple noise removal by removing data points (set to ``NaN``) below
     an adaptively estimated noise floor [1]_.
```
Comment (leewujung) [inline docs/source/process.rst:18]: Add a bullet point:
```
- Compute mean volume backscattering strength (MVBS) based on either the number of pings and sample intervals (the ``range_bin`` dimension in the data set) or a specified ping time interval and range interval in physics units (seconds and meters).
```

File: docs/source/process.rst [hunk 9]
```diff
@@ -22,75 +21,44 @@ Functionality
 
 - EK80 and EA640 broadband echosounders:
 
-  - calibration based on pulse compression output in the
+  - Calibration based on pulse compression output in the
     form of average over frequency.
 
+  - The same noise removal and MVBS functionality available
```
Comment (leewujung) [inline docs/source/process.rst:27]: change `MVBS functionality` to `MVBS computation functionality`

File: docs/source/process.rst [hunk 10]
```diff
@@ -22,75 +21,44 @@ Functionality
...
+   Sv_clean = ep.preprocess.remove_noise(          # obtain a denoised Sv Dataset
+      Sv,
+      ping_num=5,
+      range_bin_num=30
+   )
+   MVBS = ep.preprocess.get_MVBS(Sv_clean)         # obtain MVBS from denoised Sv
+
+The functions in the ``calibrate`` subpackage take in an ``EchoData`` object,
+which is essentially a container for multiple xarray ``Datasets``,
+and return a single xarray ``Dataset`` (either Sv or Sp).
```
Comment (leewujung) [inline docs/source/process.rst:48]: "(either Sv or Sp)" is not correct. The output dataset contains both Sv and range, or Sp and range.

Change this to:
```
and return a single xarray ``Dataset`` containing the calibrated backscatter quantities and the samples' corresponding range in meters.
```

File: docs/source/process.rst [hunk 11]
```diff
@@ -22,75 +21,44 @@ Functionality
...
+The Sv ``Dataset`` can then be passed into the functions provided by the
+``preprocess`` subpackage which all return a single processed ``Dataset``.
```
Comment (leewujung) [inline docs/source/process.rst:50]: Change this to a more top-down style:
```
The inputs and outputs of all functions in the ``preprocess`` subpackage are xarray ``Datasets``, 
with the inputs being ``Datasets`` containing ``Sv`` and ``range`` generated from calibration.
```

File: docs/source/process.rst [hunk 12]
```diff
@@ -22,75 +21,44 @@ Functionality
...
+
+These ``calibrate`` and ``preprocess`` methods do not save the calculation results to disk,
+but the returned xarray Dataset have methods like ``to_netcdf`` and ``to_zarr`` that
+can be used to save.
```
Comment (leewujung) [inline docs/source/process.rst:54]: Small change:
```
but the returned xarray ``Dataset`` can be save using native xarray methods such as ``to_netcdf`` and ``to_zarr`` ```

File: docs/source/process.rst [hunk 13]
```diff
@@ -115,33 +83,38 @@ By default, echopype uses the following for calibration:
...
+
+- AZFP: Salinity and pressure provided by the user,
   and temperature recorded at the instrument
 
 These parameters should be overwritten when they differ from the actual
 environmental condition during data collection.
-To update these parameters, simply do the following *before*
-calling ``ed.calibrate()``:
+To update these parameters, simply pass in the environmental parameters
+as a dictionary while calling ``ep.calibrate.compute_Sv()`` like so:
```
Comment (leewujung) [inline docs/source/process.rst:94]: These parameters should be overwritten --> These parameters **can be** overwritten

File: docs/source/process.rst [hunk 14]
```diff
@@ -115,33 +83,38 @@ By default, echopype uses the following for calibration:
...
-   ed.temperature = 8   # temperature in degree Celsius
-   ed.salinity = 30     # salinity in PSU
-   ed.pressure = 50     # pressure in dbar
-   ed.recalculate_environment()  # recalculate related parameters
+   environment = {
+       'temperature': 8,   # temperature in degree Celsius
+       'salinity': 30,     # salinity in PSU
+       'pressure': 50,     # pressure in dbar
+   }
+   Sv = ep.calibrate.compute_Sv(echodata, env_params=environment)
```
Comment (leewujung) [inline docs/source/process.rst:103]: change the dictionary name `environment` to `env_params` to avoid confusion

File: docs/source/process.rst [hunk 15]
```diff
@@ -165,24 +138,31 @@ from the raw data file.
...
+specify ``cal_params`` while calling the calibration functions like so:
 
 .. code-block:: python
 
-   ed.equivalent_beam_angle.loc[dict(frequency=18000)] = -18.02  # set value for 18 kHz only
+   # set all channels at once
+   calibration = {
+       'equivalent_beam_angle': [-17.47, -20.77, -21.13, -20.4 , -30]
+   }
+   Sv = ep.calibrate.compute_Sv(echodata, cal_params=calibration)
```
Comment (leewujung) [inline docs/source/process.rst:157]: Change `calibration` to `cal_params`

Also I think we should demonstrate passing an xr.DataArray with a frequency dimension instead of a list without any frequency labels. 
@ngkavin can you test constructing an xr.DataArray using coordinates from `echodata.beam.frequency` and test passing it in? It should just work, but if it does not let me know.

File: docs/source/process.rst [hunk 16]
```diff
@@ -165,24 +138,31 @@ from the raw data file.
...
 
 .. code-block:: python
 
-   ed.equivalent_beam_angle = [-17.47, -20.77, -21.13, -20.4 , -30]  # set all channels at once
+   echodata.beam.equivalent_beam_angle.loc[dict(frequency=18000)] = 18.02  # set value for 18 kHz only
 
-Make sure you use ``ed.equivalent_beam_angle.frequency`` to check
+Make sure you use ``echodata.beam.equivalent_beam_angle.frequency`` to check
 the sequence of the frequency channels first, and always double
 check after setting these parameters!
```
Comment (leewujung) [inline docs/source/process.rst:167]: The construction of the xr.DataArray from the above comment should allow removing this sentence all together. We actually need a check in the calibration function to take care of this, but let's delay that to later release.

File: docs/source/process.rst [hunk 17]
```diff
@@ -203,7 +183,7 @@ References
 
 .. [4] Demer DA, Berger L, Bernasconi M, Bethke E, Boswell K, Chu D, Domokos R,
    et al. (2015) Calibration of acoustic instruments. `ICES Cooperative Research Report No.
-   326. 133 pp. <https://doi.org/10.17895/ices.pub.5494>`_
+   1.   133 pp. <https://doi.org/10.17895/ices.pub.5494>`_
```
Comment (leewujung) [inline docs/source/process.rst:186]: Why is this changed to No. 1 ?  It is No. 326!

**Review by leewujung — CHANGES_REQUESTED** [review]
I requested some text changes and a code check. 

Please be careful with your editor if it changes numbers automatically like in the last comment.

File: docs/source/process.rst (hunk 17 above)
Comment (ngkavin) [reply docs/source/process.rst]: Thanks for pointing this out. I was aware that my editor was doing this to this specific number for some reason (apparently every time I delete a line of text), but I missed it in the last commit.

File: docs/source/process.rst [hunk 18]
```diff
@@ -45,13 +53,15 @@ The steps for performing these analyses are summarized below.
...
+quantities and the samples' corresponding range in meters.
+The inputs and outputs of all functions in the ``preprocess``
+subpackage are xarray ``Datasets``, with the inputs being ``Datasets``
+containing ``Sv`` and ``range`` generated from calibration.
 
 These ``calibrate`` and ``preprocess`` methods do not save the calculation results to disk,
-but the returned xarray Dataset have methods like ``to_netcdf`` and ``to_zarr`` that
-can be used to save.
+but the returned xarray ``Dataset`` can be saved using native xarray methods
+such as ``to_netcdf`` and ``to_zarr``
```
Comment (leewujung) [inline docs/source/process.rst:65]: missing a period here.

File: docs/source/process.rst (hunk 11 above)
Comment (leewujung) [reply docs/source/process.rst]: Ugh, sorry, can you change everything to singular form?

**Comment by leewujung** [comment]
@ngkavin thanks for the quick turnaround! I'll merge this now.

**Review by leewujung — APPROVED** [review]
